src/System.Private.ServiceModel/src/System/ServiceModel/Security/CryptoHelper.cs

```diff
@@ -0,0 +1,364 @@
+using System.Runtime;
+
+namespace System.ServiceModel.Security
+{
+    static class CryptoHelper
+    {
+        private static byte[] s_emptyBuffer;
+
+        // Issue #31 in progress
+        // static readonly RandomNumberGenerator random = new RNGCryptoServiceProvider();
+
+        enum CryptoAlgorithmType
+        {
+            Unknown,
+            Symmetric,
+            Asymmetric
+        }
+
+        internal static byte[] EmptyBuffer
+        {
+            get
+            {
+                if (s_emptyBuffer == null)
+                {
+                    byte[] tmp = new byte[0];
+                    s_emptyBuffer = tmp;
+                }
+                return s_emptyBuffer;
+            }
+        }
+
+        internal static HashAlgorithm NewSha1HashAlgorithm()
+        {
+            return CryptoHelper.CreateHashAlgorithm(SecurityAlgorithms.Sha1Digest);
+        }
+
+        internal static HashAlgorithm NewSha256HashAlgorithm()
+        {
+            return CryptoHelper.CreateHashAlgorithm(SecurityAlgorithms.Sha256Digest);
+        }
+
+        [SuppressMessage("Microsoft.Security.Cryptography", "CA5354:DoNotUseSHA1", Justification = "Cannot change. Required as SOAP spec requires supporting SHA1.")]
+        internal static HashAlgorithm CreateHashAlgorithm(string digestMethod)
+        {
+            throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+
+            //object algorithmObject = CryptoAlgorithms.GetAlgorithmFromConfig(digestMethod);
+            //if (algorithmObject != null)
+            //{
+            //    HashAlgorithm hashAlgorithm = algorithmObject as HashAlgorithm;
+            //    if (hashAlgorithm != null)
+            //        return hashAlgorithm;
+            //    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new MessageSecurityException(SR.GetString(SR.CustomCryptoAlgorithmIsNotValidHashAlgorithm, digestMethod)));
+            //}
+
+            //switch (digestMethod)
+            //{
+            //    case SecurityAlgorithms.Sha1Digest:
+            //        if (SecurityUtilsEx.RequiresFipsCompliance)
+            //            return new SHA1CryptoServiceProvider();
+            //        else
+            //            return new SHA1Managed();
+            //    case SecurityAlgorithms.Sha256Digest:
+            //        if (SecurityUtilsEx.RequiresFipsCompliance)
+            //            return new SHA256CryptoServiceProvider();
+            //        else
+            //            return new SHA256Managed();
+            //    default:
+            //        throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new MessageSecurityException(SR.GetString(SR.UnsupportedCryptoAlgorithm, digestMethod)));
+
+            //}
+        }
+
+        [SuppressMessage("Microsoft.Security.Cryptography", "CA5354:DoNotUseSHA1", Justification = "Cannot change. Required as SOAP spec requires supporting SHA1.")]
```

Comment:
For later - Remove the FxCop suppression, but it might be useful to keep the comment around in case in future we are questioned about why SHA1 is being used